proposals/solid-oidc/primer.bs

```diff
@@ -0,0 +1,745 @@
+}
+```
+
+#### 14. Generates a DPoP Header
+
+Now that we generated a private key for the client, we need to generate the DPoP header. To do so, we create a (JSON Web Token)[https://jwt.io/introduction/] and sign it using the key we generated.
+
+Our token could look like the following (you can decode the token using https://jwt.io):
+```
+eyJhbGciOiJFUzI1NiIsInR5cCI6ImRwb3Arand0IiwiandrIjp7Imt0eSI6IkVDIiwia2lkIjoiZkJ1STExTkdGbTQ4Vlp6RzNGMjVDOVJmMXYtaGdEakVnV2pEQ1BrdV9pVSIsInVzZSI6InNpZyIsImFsZyI6IkVDIiwiY3J2IjoiUC0yNTYiLCJ4IjoiOWxlT2gxeF9IWkhzVkNScDcyQzVpR01jek1nUnpDUFBjNjBoWldfSFlLMCIsInkiOiJqOVVYcnRjUzRLVzBIYmVteW1vRWlMXzZ1cko0TFFHZXJQZXVNaFNEaV80In19.eyJodHUiOiJodHRwczovL3NlY3VyZWF1dGguZXhhbXBsZS90b2tlbiIsImh0bSI6InBvc3QiLCJqdGkiOiI0YmEzZTllZi1lOThkLTQ2NDQtOTg3OC03MTYwZmE3ZDNlYjgiLCJpYXQiOjE2MDMzMDYxMjgsImV4cCI6MTYwMzMwOTcyOH0.2lbgLoRCkj0MsDc9BpquoaYuq0-XwRf_URdXru2JKrVzaWUqQfyKRK76_sQ0aJyVwavM3pPswLlHq2r9032O7Q
+```
+
+Token Header:
+```json
+{
+  "alg": "ES256",
+  "typ": "dpop+jwt",
+  "jwk": {
+    "kty": "EC",
+    "kid": "2i00gHnREsMhD5WqsABPSaqEjLC5MS-E98ykd-qtF1I",
+    "use": "sig",
+    "alg": "EC",
+    "crv": "P-256",
+    "x":"N6VsICiPA1ciAA82Jhv7ykkPL9B0ippUjmla8Snr4HY",
+    "y":"ay9qDOrFGdGe_3hAivW5HnqHYdnYUkXJJevHOBU4z5s",
+  }
+}
+```
+ - `"alg": "ES256"`: The signing algorithm used in this token. In this case, `ES256` because we generated the keys using eliptic curves.
+ - `"typ": "dpop+jwt"`: The type of token. All DPoP Tokens should have a type of "dpop+jwt"
+ - `"jwk": { "kty": "EC" ... }`: The client's public key.
+
+Token Body:
+```json
+{
+  "htu": "https://secureauth.example/token",
+  "htm": "post",
+  "jti": "4ba3e9ef-e98d-4644-9878-7160fa7d3eb8",
+  "iat": 1603306128,
+  "exp": 1603309728
+}
+```
+ - `"htu": "https://secureauth.example/token"`: htu limits the token for use only on the given url.
+ - `"htm": "post"`: htm limits the token for use only on a specific http method, in this case `POST`.
+ - `"jti": "4ba3e9ef-e98d-4644-9878-7160fa7d3eb8"`: jti is a unique identifier for the DPoP token that can optionally be used by ther server to defend against replays
+ - `"iat": 1603306128`: The date the token was issued, in this case October 21, 2020 14:48:48 GMT.
+ - `"exp": 1603309728`: The date this token will expire, in this case October 21, 2020 15:48:48 GMT.
+
+ #### 15. Generates Basic Authentication
+
+When sending a token request, we need to use Basic Authentication. This works by combining the client id and client secret in a string separated by a colon then base64 encoding it.
+
+For example, our client id is `https://decentphotos.com/webid#this` and our client secret is `thisIsASecret`, so the full string would be `https://decentphotos.com/webid#this:thisIsASecret` which is `aHR0cHM6Ly9kZWNlbnRwaG90b3MuY29tL3dlYmlkI3RoaXM6dGhpc0lzQVNlY3JldA==` when base64 encoded.
+
+NOTE: Based on the spec, it is currently unknown how an app with an app WebId gets its client secret.
```

Comment:
As stated above, a client secret is irrelevant in this case: there will not be one.